api/v1alpha1/nnfsystemstorage_types.go

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2024 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	dwsv1alpha2 "github.com/DataWorkflowServices/dws/api/v1alpha2"
+	"github.com/DataWorkflowServices/dws/utils/updater"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type NnfSystemStorageComputesTarget string
+
+const (
+	ComputesTargetAll     NnfSystemStorageComputesTarget = "all"
+	ComputesTargetEven    NnfSystemStorageComputesTarget = "even"
+	ComputesTargetOdd     NnfSystemStorageComputesTarget = "odd"
+	ComputesTargetPattern NnfSystemStorageComputesTarget = "pattern"
+)
+
+// NnfSystemStorageSpec defines the desired state of NnfSystemStorage
+type NnfSystemStorageSpec struct {
+	// SystemConfiguration is an object reference to the SystemConfiguration resource to use. If this
+	// field is empty, name: default namespace: default is used.
+	SystemConfiguration corev1.ObjectReference `json:"systemConfiguration,omitempty"`
+
+	// ExludeRabbits is a list of Rabbits to exclude from the Rabbits in the SystemConfiguration
+	ExcludeRabbits []string `json:"excludeRabbits,omitempty"`
+
+	// IncludeRabbits is a list of Rabbits to use rather than getting the list of Rabbits from the
+	// SystemConfiguration
+	IncludeRabbits []string `json:"includeRabbits,omitempty"`
+
+	// ExcludeComputes is a list of compute nodes to exclude from the the compute nodes listed in the
+	// SystemConfiguration
+	ExcludeComputes []string `json:"excludeComputes,omitempty"`
+
+	// IncludeComputes is a list of computes nodes to use rather than getting the list of compute nodes
+	// from the SystemConfiguration
+	IncludeComputes []string `json:"includeComputes,omitempty"`
```

Comment:
By default, does this just grab random rabbits and computes from the SystemConfiguration if you don't use these include/exclude fields?

Reply:
Makes sense. I think you should make sure to document that somewhere. If not here, then definitely in the nnf docs.

Reply:
Yeah, I'm going to do a write up on how to use this in the nnf docs.